### PR TITLE
chore(argo-events): update argo-events to 1.2.3

### PR DESCRIPTION
**Pull request by smcavallo**
Signed-off-by: smcavallo <smcavallo@hotmail.com>

https://github.com/argoproj/argo-events/releases
https://github.com/nats-io/nats-streaming-server/releases

Checklist:

* [X] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X] I have signed the CLA and the build is green.
* [X] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by vladlosev** [comment]
Looks like it's not enough to just bump the version. I am getting validation errors when trying to apply crds from your branch:

> (base) vlosev-mbp:argo-helm vlad$ kubectl apply -f charts/argo-events/crds/eventbus-crd.yml
error: error validating "charts/argo-events/crds/eventbus-crd.yml": error validating data: [ValidationError(CustomResourceDefinition.spec): unknown field "version" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec, ValidationError(CustomResourceDefinition.spec): missing required field "versions" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec]; if you choose to ignore these errors, turn validation off with --validate=false

File: charts/argo-events/values.yaml [hunk 1]
```diff
@@ -46,21 +46,21 @@ singleNamespace: true
...
   eventsourceImage: eventsource
 
 eventbusController:
   name: eventbus-controller
   image: eventbus-controller
-  tag: v1.0.0
+  tag: v1.2.0
   replicaCount: 1
-  natsStreamingImage: nats-streaming:0.17.0
+  natsStreamingImage: nats-streaming:0.20.0
```
Comment (whynowy) [inline charts/argo-events/values.yaml:65]: I myself never tried this version, prefer to keep the same version as before.

File: charts/argo-events/values.yaml [hunk 2]
```diff
@@ -46,21 +46,21 @@ singleNamespace: true
 sensorController:
   name: sensor-controller
   image: sensor-controller
-  tag: v1.0.0
+  tag: v1.2.0
```
Comment (whynowy) [inline charts/argo-events/values.yaml:49]: Can you upgrade to `v1.2.1`?

**Review by whynowy — COMMENTED** [review]
https://github.com/argoproj/argo-helm/pull/518/files

Also, please update the controller deployments with probe and security context, refer to https://github.com/argoproj/argo-helm/pull/518/files.

File: charts/argo-events/values.yaml (hunk 2 above)
Comment (whynowy) [reply charts/argo-events/values.yaml:49]: Sorry, do `v1.2.2` please.

Comment (davidkarlsen) [inline charts/argo-events/values.yaml:49]:
```suggestion
  tag: v1.2.2
```

**Comment by smcavallo** [comment]
I tried to incorporate all the changes and squash into a single commit

Upgrade to 1.2.2
- add liveness probe and readiness probe
See - https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
- support for securityContext
See - https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
- support for nodeSelector, tolerations, affinity
See - https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/
- update CustomResourceDefinition api version (deprecated in 1.16)
See - https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/

File: charts/argo-events/templates/eventsource-controller-deployment.yaml [hunk 3]
```diff
@@ -35,7 +35,35 @@ spec:
...
+            periodSeconds: 3
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: 8081
+            initialDelaySeconds: 3
+            periodSeconds: 3
+      {{- with .Values.securityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
```
Comment (davidkarlsen) [inline charts/argo-events/templates/eventsource-controller-deployment.yaml:52]: with nindent this is typically on the same line: https://banzaicloud.com/blog/creating-helm-charts-part-2/

File: charts/argo-events/templates/eventsource-controller-deployment.yaml [hunk 4]
```diff
@@ -35,7 +35,35 @@ spec:
...
+      {{- end }}
       {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+{{- if .Values.eventsourceController.nodeSelector }}
```
Comment (davidkarlsen) [inline charts/argo-events/templates/eventsource-controller-deployment.yaml:58]: these can be shortened by using the `with` style, however both are possible and will work, but I'd favour using with

File: charts/argo-events/templates/eventsource-controller-deployment.yaml (hunk 3 above)
Comment (smcavallo) [reply charts/argo-events/templates/eventsource-controller-deployment.yaml]: @davidkarlsen - much more readable - I have made this change

**Comment by smcavallo** [comment]
@davidkarlsen @whynowy - anything else you need from me?  anything else I should do here?

**Comment by lukasmrtvy** [comment]
@smcavallo maybe you can bump it to 1.2.3

**Comment by smcavallo** [comment]
@lukasmrtvy - it has been bumped to 1.2.3

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**